internal/store/store.go

```diff
@@ -55,17 +56,46 @@ type BlockStore struct {
 	mtx    cmtsync.RWMutex
 	base   int64
 	height int64
+
+	blocksDeleted      int64
+	compact            bool
+	compactionInterval int64
+
+	logger log.Logger
+}
+
+type BlockStoreOption func(*BlockStore)
+
+// WithCompaction sets the compaciton parameters.
+func WithCompaction(compact bool, compactionInterval int64) BlockStoreOption {
+	return func(bs *BlockStore) {
+		bs.compact = compact
+		bs.compactionInterval = compactionInterval
+	}
+}
+
+// WithLogger sets the logger for the blockstore.
```

Comment:
```suggestion
// WithLogger sets the logger for the BlockStore.
```
just to make it consistent with other comments